code/modules/atmospherics/machinery/components/unary_devices/vent_pump.dm

```diff
@@ -169,9 +169,20 @@
 	//you cannot unassign from an area we never were assigned to
 	if(isnull(target_area) || assigned_area != target_area)
 		return
+	if(src in assigned_area.excluded_vents)
+		assigned_area.excluded_vents -= src
```

Comment:
there's no reason to check beforehand, it basically does the exact same thing as -= except it doesn't actually remove the thing.

Reply:
good point, my stupid brain almost thought of it itself